components/EmotionListAndFilter/useScreenSize.js

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+
+export default function useScreenSize() {
+  const [screenSize, setScreenSize] = useState({
+    width: 0,
+    height: 0,
+  });
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
```

Comment:
a useEffect is only run in the client, therefore you don't need to check if window does exist.

Reply:
usually, hooks are placed in a folder `lib/hooks/` not in the components folder.